src/Network/Response.php

```diff
@@ -686,16 +688,36 @@ public function header($header = null, $value = null)
                 list($header, $value) = explode(':', $header, 2);
             }
 
-            if ($this->hasHeader($header)) {
-                $header = $this->headerNames[strtolower($header)];
+            $lower = strtolower($header);
+            if (array_key_exists($lower, $this->headerNames)) {
+                $header = $this->headerNames[$lower];
             } else {
-                $this->headerNames[strtolower($header)] = $header;
+                $this->headerNames[$lower] = $header;
             }
 
-            $this->headers[$header] = is_array($value) ? array_map('trim', $value) : trim($value);
+            $this->headers[$header] = is_array($value) ? array_map('trim', $value) : [trim($value)];
+        }
+
+        return $this->getSimpleHeaders();
```

Comment:
This fixes the return of `header()` not returning a simple dictionary when headers are set with the PSR7 methods.